node/bft/src/sync/mod.rs

```diff
@@ -118,6 +118,18 @@ impl<N: Network> Sync<N> {
             }
         }));
 
+        // Start the pending queue expiration loop.
+        let self_ = self.clone();
+        self.spawn(async move {
+            loop {
+                // Sleep briefly.
+                tokio::time::sleep(Duration::from_millis(MAX_FETCH_TIMEOUT_IN_MS)).await;
+
+                // Remove the expired pending transmission requests.
+                self_.pending.clear_expired_callbacks();
```

Comment:
we might want this to be called within a blocking task

Reply:
Resolved with https://github.com/AleoHQ/snarkOS/pull/3135/commits/a39996d213f0a3e9c2556bb305d1fce2e00bc1d6